src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1034,6 +1034,7 @@ free_request(struct sol_http_request *request)
         switch (param->type) {
         case SOL_HTTP_PARAM_POST_DATA:
             free((char *)param->value.data.filename.data);
+            break;
```

Comment:
I guess that @ceolin is the one who can validate this - but I guess that current code is correct as well: code after `FOREACH` does basically the same for `request->param` - frees unconditionally `param.value.key_value.value.data` and `param.value.key_value.key.data`, and if `SOL_HTTP_PARAM_POST_DATA`, frees `param.value.data.filename.data`.